Scripts/RemediationScripts/Remediate-SetSQLServerMinReqTLSVersion.ps1

```diff
@@ -0,0 +1,962 @@
+            $sqlServerResourcesFromFile = Import-Csv -LiteralPath $FilePath
+            $validsqlServerResources = $sqlServerResourcesFromFile | Where-Object { ![String]::IsNullOrWhiteSpace($_.ServerName) }
+    
+            $validsqlServerResources | ForEach-Object {
+                $resourceGroupName = $_.ResourceGroupName        
+                $serverName = $_.ServerName               
+
+                try
+                {
+                    $sqlServerResources += (Get-AzSqlServer -ResourceGroupName $resourceGroupName -ServerName $serverName -ErrorAction SilentlyContinue) 
+
+                }
+                catch
+                {
+                    Write-Host "Error fetching Server:   - $($serverName). Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+                    Write-Host "Skipping this Server..." -ForegroundColor $([Constants]::MessageType.Warning)
+                    Write-Host $([Constants]::SingleDashLine)
+                }
+            }
+        }
+    }
+    
+    $totalsqlServerResources = ($sqlServerResources | Measure-Object).Count
+
+    if ($totalsqlServerResources -eq 0)
+    {
+        Write-Host "No SQL Servers found. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::DoubleDashLine)
+        break
+    }
+
+    Write-Host "Found $($totalsqlServerResources) SQL Server(s)." -ForegroundColor $([Constants]::MessageType.Update)
+    Write-Host $([Constants]::SingleDashLine)
+ 
+     
+    # Includes SQL Servers where minimal required TLS version is set  
+    $sqlServersWithReqMinTLSVersion = @()
+
+    # Includes SQL Servers where minimal required TLS version is not set   
+    $sqlServersWithoutReqMinTLSVersion = @()
+
+    # Includes SQL Servers that were skipped during remediation. There were errors remediating them.
+    $sqlServersSkipped = @()
+
+     
+    Write-Host "`n[Step 3 of 5] Fetching SQL Servers with (s)..."
+    Write-Host $([Constants]::SingleDashLine)
+    Write-Host "Separating SQL Server(s) for which TLS is less than required TLS version ..." -ForegroundColor $([Constants]::MessageType.Info)
+    Write-Host $([Constants]::SingleDashLine)
+    $sqlServerResources | ForEach-Object {
+        $sqlServer = $_        
+        if($_.MinimalTlsVersion -lt $requiredMinTLSVersion) 
+        {
+            $sqlServersWithoutReqMinTLSVersion +=  $sqlServer | Select-Object @{N='ServerName';E={$_.ServerName}},
+                                                                        @{N='ResourceGroupName';E={$_.ResourceGroupName}},
+                                                                        @{N='Location';E={$_.Location}},
+                                                                        @{N='ServerVersion';E={$_.ServerVersion}},
+                                                                        @{N='MinimalTlsVersion';E={$_.MinimalTlsVersion}}
+        }
```

Comment:
what is sql does have min tls set, shouldn't they be added as skipped resources?

Reply:
added to skipped